frontend/src/lib/search-utils.ts

```diff
@@ -22,32 +22,61 @@ export const handleStandardComparison = (valueOne: string, valueTwo: string, ope
   }
 }
 
-// for a given displayVersion string there is a distribution value a ' ' and a semver value, here we divide them and take the semver
-export const handleSemverOperatorComparison = (versionOne: string, versionTwo: string, operator: SearchOperator) => {
-  // Semver coerces the version to a valid semver version if possible, otherwise it returns the original value
-  const coercedVersionOne = semver.valid(semver.coerce(versionOne)) ?? versionOne
-  const coercedVersionTwo = semver.valid(semver.coerce(versionTwo)) ?? versionTwo
+export const handleSemverOperatorComparison = (
+  versionToCompare: string,
+  inputVersion: string,
+  operator: SearchOperator
+) => {
+  const coerceVersion = (version: string) => semver.valid(semver.coerce(version)) ?? version
+  const coercedInputVersion = coerceVersion(inputVersion)
+  const coercedVersionToCompare = coerceVersion(versionToCompare)
 
-  const validInputSemvers = !!semver.valid(coercedVersionOne) && !!semver.valid(coercedVersionTwo)
-  if (!validInputSemvers) {
-    if (operator === SearchOperator.NotEquals) {
-      return true
+  // If the inputVersion is not a valid semver, but the coerced version is, we can use the incomplete version to create a range
+  if (!semver.valid(inputVersion) && semver.valid(semver.coerce(inputVersion))) {
+    const isMajorVersionAlone = !inputVersion.includes(
+      `${semver.coerce(inputVersion)?.major}.${semver.coerce(inputVersion)?.minor}`
+    )
+    const topRange = isMajorVersionAlone
+      ? semver.inc(coercedInputVersion, 'major')
+      : semver.inc(coercedInputVersion, 'minor')
+
+    const equalsValueRange = `${coercedInputVersion} - ${topRange}`
```

Comment:
I think this would include `topRange`, but it should be excluded.

I think we can just use "X" or "X.Y" as a range. See [X-Ranges 1.2.x 1.X 1.2.* *](https://github.com/npm/node-semver?tab=readme-ov-file#x-ranges-12x-1x-12-): "A partial version range is treated as an X-Range, so the special character is in fact optional."

```
semver.satisfies("5.0.0", "4.0.0 - 5.0.0")
true
semver.satisfies("5.0.0", "4")
false
semver.satisfies("4.14.50", "4.14")
true
semver.satisfies("4.17.3", "4")
true
```